.github/workflows/tests.yml

```diff
@@ -34,27 +27,96 @@ jobs:
       with:
         context: .
         push: false
-        tags: appwrite-dev
+        tags: ${{ env.IMAGE }}
         load: true
         cache-from: type=gha
         cache-to: type=gha,mode=max
+        outputs: type=docker,dest=/tmp/${{ env.IMAGE }}.tar
         build-args: |
           DEBUG=false
           TESTING=true
           VERSION=dev
 
-    - name: Start Appwrite
-      run:  |
-        docker compose up -d
-        sleep 30
+    # - name: Pull Docker Images
+    #   run: docker compose pull --ignore-pull-failures --ignore-buildable
+    
+    # - name: List Files and images
+    #   run: |
+    #     docker compose config --images | sort -u
+    #     docker save $(docker compose config --images | sort -u) -o /tmp/allinone.tar
+
+    - name: Cache Docker Image
+      uses: actions/cache@v3
+      with:
+        key: ${{ env.IMAGE }}-${{ github.sha }}
+        path: /tmp/${{ env.IMAGE }}.tar
+
+    # - name: Upload Docker Image
+    #   uses: actions/upload-artifact@v3
+    #   with:
+    #     name: ${{ env.IMAGE }}
+    #     path: /tmp/${{ env.IMAGE }}.tar
+
+  unit_test:
+    name: Unit Test
+    runs-on: ubuntu-latest
+    needs: setup
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+    
+      - name: Load Cache
+        uses: actions/cache@v3
+        with:
+          key: ${{ env.IMAGE }}-${{ github.sha }}
+          path: /tmp/${{ env.IMAGE }}.tar
+          fail-on-cache-miss: true
+      
+      - name: Load and Start Appwrite
+        run:  |
+          docker load --input /tmp/${{ env.IMAGE }}.tar
+          docker compose up -d
+          sleep 10
+
+      - name: Doctor
+        run: docker compose exec -T appwrite doctor
+
+      - name: Environment Variables
+        run:  docker compose exec -T appwrite vars
+
+      - name: Run Unit Tests
+        run: docker compose exec appwrite test /usr/src/code/tests/unit 
+
+  e2e_test:
+    name: E2E Test
+    runs-on: ubuntu-latest
+    needs: setup
+    strategy:
+      fail-fast: false
+      matrix:
+        services: [Account,Avatars,Console,Databases,Functions,GraphQL,Health,Locale,Projects,Realtime,Storage,Teams,Users,Webhooks]
```

Comment:
Can we rename this to `service`? That way later we can do `Services/${{matrix.service}}`, sounds more verbose than having plural there.

Reply:
done